tools/slicec-cs/src/decoding.rs

```diff
@@ -210,161 +210,70 @@ decoder.DecodeResult(
-                    remove_optional_modifier_from(incoming_element_type_string),
-                    if matches!(primitive, Primitive::Bool) {
-                        "checkElement: SliceDecoder.CheckBoolValue"
-                    } else {
-                        ""
-                    }
-                )
-            }
-            Types::Enum(enum_def) if enum_def.underlying.is_some() && enum_def.fixed_wire_size().is_some() => {
-                if enum_def.is_unchecked {
-                    write!(
-                        code,
-                        "decoder.DecodeSequence<{}>()",
-                        remove_optional_modifier_from(incoming_element_type_string),
-                    )
-                } else {
-                    write!(
-                        code,
-                        "\
-decoder.DecodeSequence(
-    ({enum_type} e) => _ = {underlying_extensions_class}.As{name}(({underlying_type})e))",
-                        enum_type = incoming_element_type_string,
-                        underlying_extensions_class = enum_def.escape_scoped_identifier_with_suffix(
-                            &format!(
-                                "{}Extensions",
-                                enum_def.get_underlying_cs_type().to_cs_case(Case::Pascal)
-                            ),
-                            namespace,
-                        ),
-                        name = enum_def.cs_identifier(Case::Pascal),
-                        underlying_type = enum_def.get_underlying_cs_type(),
-                    );
-                }
+        Types::Enum(enum_def) if enum_def.fixed_wire_size().is_some() && !uses_bit_sequence => {
+            if enum_def.is_unchecked {
+                builder.set_type_argument(remove_optional_modifier_from(element_type_string));
+            } else {
+                let underlying_type = enum_def.get_underlying_cs_type();
+                let enum_name = enum_def.cs_identifier(Case::Pascal);
+                let underlying_extensions_class = enum_def.escape_scoped_identifier_with_suffix(
+                    &(underlying_type.to_cs_case(Case::Pascal) + "Extensions"),
+                    namespace,
+                );
+                let decode_func = format!("({element_type_string} e) => _ = {underlying_extensions_class}.As{enum_name}(({underlying_type})e)");
+                builder.add_argument(decode_func);
             }
-            _ => {
-                write!(
-                    code,
-                    "\
-decoder.DecodeSequence(
-    {})",
-                    decode_func(element_type, namespace, encoding, false).indent(),
-                )
+        }
+        _ => {
+            if has_cs_type_attribute {
+                uses_sequence_factory = true;
+                builder.add_argument(format!("sequenceFactory: (size) => new {sequence_type}(size)"));
             }
+            builder.add_argument(decode_func(element_type, namespace, encoding, false).indent());
```

Comment:
Is this indent still correct?

Reply:
At least it doesn't produce a difference in the generated code.
I was pretty careful to avoid changing behavior.